openapi-client-idl-plugin/src/main/java/io/ballerina/openapi/idl/client/OpenAPIClientIDLPlugin.java

```diff
@@ -0,0 +1,450 @@
+                    return;
+                }
+                String moduleName = "openapi_client";
+                ModuleId moduleId = ModuleId.create(moduleName, idlSourceContext.currentPackage().packageId());
+                List<DocumentConfig> documents = new ArrayList<>();
+
+                genSrcFiles.forEach(genSrcFile -> {
+                    DocumentId documentId = DocumentId.create(genSrcFile.getFileName(), moduleId);
+                    DocumentConfig documentConfig = DocumentConfig.from(
+                            documentId, genSrcFile.getContent(), genSrcFile.getFileName());
+                    documents.add(documentConfig);
+
+                });
+
+                ModuleDescriptor moduleDescriptor = ModuleDescriptor.from(
+                        ModuleName.from(idlSourceContext.currentPackage().packageName(), moduleName),
+                        idlSourceContext.currentPackage().descriptor());
+                ModuleConfig moduleConfig =
+                        ModuleConfig.from(moduleId, moduleDescriptor, documents, new ArrayList<>(), null,
+                                new ArrayList<>());
+                NodeList<AnnotationNode> annotations = getAnnotationNodes(idlSourceContext);
+                //only pass openapi related annotation node.
+                List<AnnotationNode> openapiAnnot = filterOpenAPIAnnotation(annotations);
+
+                idlSourceContext.addClient(moduleConfig, openapiAnnot.isEmpty() ? NodeFactory.createEmptyNodeList() :
+                        NodeFactory.createNodeList(openapiAnnot));
+            } catch (BallerinaOpenApiException e) {
+                Constants.DiagnosticMessages error = Constants.DiagnosticMessages.PARSER_ERROR;
+                DiagnosticInfo diagnosticInfo = new DiagnosticInfo(error.getCode(), e.getMessage(),
+                        error.getSeverity());
+                Diagnostic diagnostic = DiagnosticFactory.createDiagnostic(diagnosticInfo,
+                        idlSourceContext.clientNode().location());
+                idlSourceContext.reportDiagnostic(diagnostic);
+            } catch (IOException | FormatterException e) {
+                Constants.DiagnosticMessages error = Constants.DiagnosticMessages.ERROR_WHILE_GENERATING_CLIENT;
+                reportDiagnostic(idlSourceContext, error, idlSourceContext.clientNode().location());
+            }
+        }
+
+        /**
+         * This is for checking the given file can handle via OpenAPI tool.
+         */
+        private static boolean isOpenAPI(Path oasPath) {
+            try {
+                if (!(oasPath.toString().endsWith(".yaml") || oasPath.toString().endsWith(".json") ||
+                        oasPath.toString().endsWith(".yml"))) {
+                    return false;
+                }
+                String content = Files.readString(oasPath);
+                return content.startsWith("openapi:") | content.startsWith("{\n" +
+                        "  \"openapi\":");
```

Comment:
Is this logic correct? What if we have more new lines or any header in the beginning of the file?

Reply:
Don't we have any API in the openAPI parser to detect if a given file is an openAPI contract (or not)?

Reply:
Currently, we have only one API from swagger to check given YAML is a valid spec or not.  Using these APIs it populates OpenAPI models for the valid yaml. 
During the meeting, it was suggested to implement  `canHandle` method at a low cost. because our complete validation goes to handle via perform a method and return spec validation within perform a method to avoid conflict with other plugins. Therefore suggested having a simple regex function for checking basic mandataries including "openapi:". 

Also, have the same thoughts and concerns if the given YAML has some commented line in the header. went through some samples and couldn't find the scenarios, but will try to optimize this logic to cover the above concern :)